chisel/src/logger.rs

```diff
@@ -0,0 +1,30 @@
+static mut LOG_LEVEL: i32 = 0;
+
+#[macro_export]
+macro_rules! chisel_debug {
+    ($lvl:expr, $($arg:tt)*) => {
+        crate::logger::Logger::with_global_level().log($lvl, &format!($($arg)*));
+    }
+}
+
+/// Simple logging utility struct.
+pub struct Logger(i32);
+
+impl Logger {
+    pub fn with_global_level() -> Self {
+        unsafe { Logger(LOG_LEVEL) }
+    }
+
+    pub fn log<T: AsRef<str>>(&self, level: i32, message: T) {
+        if self.0 >= level {
+            eprintln!("{}", message.as_ref());
```

Comment:
Why is this going to stderr?

Reply:
Because it is a debug message.

Reply:
I guess its debatable, but most tools which do not output consumable data onto stdout, will use stdout for debug.